c7n/resources/apigw.py

```diff
@@ -793,3 +793,69 @@ def process(self, resources):
                     resourceId=m['resourceId'],
                     httpMethod=m['httpMethod'],
                     patchOperations=ops)
+
+
+@resources.register('apigw-domain-name')
+class CustomDomainName(query.QueryResourceManager):
+
+    class resource_type(query.TypeInfo):
+        enum_spec = ('get_domain_names', 'items', None)
+        arn = False
+        id = name = 'domainName'
+        service = 'apigateway'
+        universal_taggable = True
+        cfn_type = 'AWS::ApiGateway::DomainName'
+        date = 'createdDate'
+
+    @classmethod
+    def get_permissions(cls):
+        return ('apigateway:GET',)
+
+    @classmethod
+    def has_arn(self):
+        return False
+
+
+@CustomDomainName.filter_registry.register('check-tls')
+class DomainNameCheckTls(Filter):
+    """ Filter API Gateway custom domains and return those with security policy != TLS 1.2
```

Comment:
this feels like its just a value filter vs needing to be a coded as its a simple attribute check.

Reply:
@kapilt i've removed the filter